### PR TITLE
BlogSettings: Implements Discussion Extension

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Details:
In this PR we implement an Extension for the `BlogSettings` entity. It's goal is to provide a high level API, to be consumed by #4397.

This can be considered as an *Adapter*, Swift style.

Needs Review: @diegoreymendez 
Thank you Diego!

### CODE REVIEW

File: WordPress/Classes/Models/BlogSettings+Discussion.swift [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+ *              This may be considered as an Adapter class, *Swift* style!
+ */
+
+extension BlogSettings
+{
+    /**
+     *  @enum   BlogSettings.CommentsAutoapproval
+     *  @brief  Enumerates all of the Comments AutoApproval settings
+     */
+    public enum CommentsAutoapproval : Int {
```
Comment (diegoreymendez) [inline WordPress/Classes/Models/BlogSettings+Discussion.swift]: Just wondering why `public`? (internal should be fine for the App classes afaik, and for unit testing you can use `@testable import`)

Comment (diegoreymendez) [inline WordPress/Classes/Models/BlogSettings+Discussion.swift]: PS: same for the other `public` keywords below.

**Comment by jleandroperez** [comment]
@diegoreymendez `public` keyword is gone. Thanks!

**Comment by diegoreymendez** [comment]
:shipit:

**Comment by jleandroperez** [comment]
Thank you @diegoreymendez !!

**Pull request merged**